geopandas/geodataframe.py

```diff
@@ -1,22 +1,24 @@
 import json
 import warnings
+from typing import Union, Callable
 
 import numpy as np
 import pandas as pd
 from pandas import DataFrame, Series
 
 from shapely.geometry import mapping, shape
 from shapely.geometry.base import BaseGeometry
+from sqlalchemy.engine.base import Connection, Engine
 
 
 from pyproj import CRS
 
 from geopandas.array import GeometryArray, GeometryDtype, from_shapely, to_wkb, to_wkt
 from geopandas.base import GeoPandasBase, is_geometry_type
-from geopandas.geoseries import GeoSeries, inherit_doc
```

Comment:
Looking at the tests, it looks like this import probably should not have been removed.

Reply:
Yes. However, I think there was a circular import when importing from `geopandas.geoseries`

Reply:
That's surprising, it seems like geodataframe would need to import geoseries, but not necessarily the other way around.

Reply:
Let me check geoseries.py

Reply:
Lines 364, 530, and 546 all import GeoDataFrame